File: images/toolchain/debian-trixie/Dockerfile.php
# Toolchain image: PHP + Composer.
#
# Generic: the artifacts are pre-downloaded (fetch-artifacts.sh) and COPYed.
ARG BASE_IMAGE=toolchain-base:debian-trixie
FROM ${BASE_IMAGE}
COPY cache/php-8.5.8-cli-linux-x86_64.tar.gz /opt/php.tar.gz
COPY cache/composer.phar /opt/composer.phar
RUN mkdir -p /opt/php && tar -xzf /opt/php.tar.gz -C /opt/php --strip-components=1 && rm /opt/php.tar.gz \
 && printf '#!/bin/sh\nexec /opt/php/bin/php /opt/composer.phar "$@"\n' > /usr/local/bin/composer \
 && chmod +x /usr/local/bin/composer
ENV PATH=/opt/php/bin:$PATH

File: images/toolchain/debian-trixie/Dockerfile.java
# Toolchain image: JDK + Gradle.
#
# Both artifacts are pre-downloaded (fetch-artifacts.sh) and COPYed. Gradle is a
# pure-java distribution, so the distro-specific part is only the JDK.
ARG BASE_IMAGE=toolchain-base:debian-trixie
FROM ${BASE_IMAGE}
COPY cache/OpenJDK25U-jdk_x64_linux_hotspot_25.0.4.1_1.tar.gz /tmp/jdk.tar.gz
COPY cache/gradle-9.7.1-bin.zip /tmp/gradle.zip
RUN mkdir -p /opt/jdk && tar -xzf /tmp/jdk.tar.gz -C /opt/jdk --strip-components=1 && rm /tmp/jdk.tar.gz \
 && unzip -q /tmp/gradle.zip -d /opt && rm /tmp/gradle.zip \
 && ln -s /opt/gradle-9.7.1 /opt/gradle
ENV JAVA_HOME=/opt/jdk PATH=/opt/jdk/bin:/opt/gradle/bin:$PATH
